libraries/chain/database/block_tasks/process_comments_cashout.cpp

```diff
@@ -25,31 +26,15 @@ void process_comments_cashout::on_apply(block_task_context& ctx)
 #endif
     dynamic_global_property_service_i& dgp_service = services.dynamic_global_property_service();
 
-    // Add all reward funds to the local cache and decay their recent rshares
-    reward_fund_service.update([&](reward_fund_object& rfo) {
-        fc::microseconds decay_rate = SCORUM_RECENT_RSHARES_DECAY_RATE;
-        rfo.recent_claims -= (rfo.recent_claims * (dgp_service.head_block_time() - rfo.last_update).to_seconds())
-            / decay_rate.to_seconds();
-        rfo.last_update = dgp_service.head_block_time();
-    });
-
     auto comments = comment_service.get_by_cashout_time();
 
-    uint128_t recent_claims = get_recent_claims(ctx, comments);
-
-    /*
-     * Payout all comments
-     *
-     * Each payout follows a similar pattern, but for a different reason.
-     * The helper only does token allocation based on curation rewards and the SCR
-     * global %, etc.
-     *
-     * Each context is used by get_rshare_reward to determine what part of each budget
-     * the comment is entitled to. Each payout is done
-     * against a reward fund state that is snapshotted before all payouts in the block.
-     */
     const auto& rf = reward_fund_service.get();
 
+    share_types total_rshares = get_total_rshares(ctx, comments);
```

Comment:
it is difficult to read and understand such `typedef` `share_types -> std::vector<share_type>`. I would prefer to see `std::vector<share_type>` type name instead of share_types.